pyanaconda/modules/payloads/source/cdn/cdn.py

```diff
@@ -29,6 +31,13 @@
 class CDNSourceModule(RepoFilesSourceModule):
     """The CDN source payload module."""
 
+    def __init__(self):
+        super().__init__()
+        self._subscription_module = None
+        # if the the Subscriptiopn DBus module is available, connect to it
+        if is_module_available(SUBSCRIPTION):
+            self._subscription_module = SUBSCRIPTION.get_proxy()
```

Comment:
This can be pretty unsafe.

Reply:
I see similar pattern (checking if module is available and then connecting to it) in a couple places. Is the source somehow specific in making this less safe ? Also, any suggestions for a more safer approach ?

Reply:
Yes, you see it in the UI, but no one communicates with the UI, so it cannot cause any deadlocks. Modules can communicate with other modules from tasks that run in threads, because that will not block the main thread that is suppose to handle DBus calls. Here you access the Boss and the Subscription module from the main thread, so the Payloads module is not able to process any new DBus calls until these calls are finished. It can work now, but it is a timing bomb.

The source module could have a new DBus method that will allow to set the type of the infrastructure or something, so the source can have all info available when it is asked for a description. You just need to update the value to the right one, but that shouldn't be a problem.

Another solution would be to rewrite these DBus calls and call them asynchronously or replace them with signal handlers, but I don't have time to work on that at this moment.

Reply:
Good point, lets simplify the whole thing. As in the end this is just about the spoke status I've all the dynamic dropped the logic from the CDN source class and instead special cased the Spoke status string generation to return "Satellite" if satellite is in use. Should be much simpler and less dynamic than before.